lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -2058,7 +2058,7 @@ bool TypeChecker::typeCheckCompletionSequence(Expr *&expr, DeclContext *DC) {
 
   // Attempt to solve the constraint system.
   SmallVector<Solution, 4> viable;
-  if (CS.solve(viable, FreeTypeVariableBinding::GenericParameters))
+  if (CS.solve(viable, FreeTypeVariableBinding::UnresolvedType))
```

Comment:
Ok, the reason why + is returned in here is because + has multiple generic overloads and UnresolvedType is more permissive then GenericParameter, so what it's matching is - 
`Swift.(file).Strideable.+ : <Self where Self : Strideable> (Self.Type) -> (Self.Stride, Self) -> Self`. I think I can tweak `typeCheckCompletionSequence` so if `CCE` is unresolved don't return it the solution.